radareorg/cutter
# Enhanced Decompiler Widget with new Dedicated Context Menu and Actions

**Your checklist for this pull request**


**Detailed description**
As part of the [Improving Decompiler Widget](https://github.com/orgs/radareorg/projects/1) project, we have made many improvements to the decompiler widget and have made a new dedicated context menu for the decompiler in the past couple of months. Most of these changes were committed to the `decompiler-refactoring` branch. This PR is to merge all of the new features to `master`.

The important changes and the corresponding PRs are listed below.
1. #2256 - Dedicated decompiler context menu (skeleton) with an action to copy selected code
2. #2260 - Breakpoint Menu and Debug Menu in the Decompiler Context Menu
3. #2265 - Comment Menu for the Decompiler Context Menu
4. #2281 - Intercept right-click event in the decompiler widget
5. #2286 - Action to rename functions in the decompiler context menu
6. #2295 - Some global variable actions and Show in action 
                   * Show in action for global variables and functions
                   * Copy address of global variable or function referenced by the cursor selection
                   * Rename global variable
7. #2351 - Fix display jumping in the widget for every decompilation. Reduce unnecessary decompilations.
8. #2352 - Xrefs action for references
9. #2357 - Edit/Rename Variables Actions for function variables
10. #2374 - Documenting code and final cleanup


**Test plan (required)**
1. Fetch the latest master of r2ghidra-dec and the current commit in the latest radare2 submodule.
2. Go through all the linked PRs and test all the features implemented by each PR. Just see if all of them are working as expected.
3. Try to check if anything else is broken by these changes using Cutter for something you normally do.





**Closing issues**
closes #1764
